### PR TITLE
Add multiple file statuses

### PR DESCRIPTION
**Pull request by MancunianSam**
I've added in a new endpoint to allow us to add multiple file statuses
as part of the new backend checks.

I've deprecated the old method and updated the tests to use the new one

I've not changed the existing mutation types because the last time I did
that it broke things.

I've removed the restrictions on the status type only allowing Upload
statuses because we don't want that now.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala [hunk 1]
```diff
@@ -15,11 +15,14 @@ object FileStatusFields {
   case class FileStatus(fileId: UUID, statusType: String, statusValue: String)
 
   case class AddFileStatusInput(fileId: UUID, statusType: String, statusValue: String)
+  case class AddMultipleFileStatusInput(statuses: List[AddFileStatusInput])
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala:18]: Should this be re-named: `AddMultipleFileStatusesInput`

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala [hunk 2]
```diff
@@ -15,11 +15,14 @@ object FileStatusFields {
...
 
   implicit val FileStatusType: ObjectType[Unit, FileStatus] = deriveObjectType[Unit, FileStatus]()
   implicit val AddFileStatusInputType: InputObjectType[AddFileStatusInput] = deriveInputObjectType[AddFileStatusInput]()
+  implicit val AddMultipleFileStatusInputType: InputObjectType[AddMultipleFileStatusInput] = deriveInputObjectType[AddMultipleFileStatusInput]()
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala:22]: Should this be re-named: `AddMultipleFileStatusesInputType`

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala [hunk 3]
```diff
@@ -15,11 +15,14 @@ object FileStatusFields {
...
 
   implicit val FileStatusInputArg: Argument[AddFileStatusInput] = Argument("addFileStatusInput", AddFileStatusInputType)
+  implicit val MultipleFileStatusInputArg: Argument[AddMultipleFileStatusInput] = Argument("addMultipleFileStatusInput", AddMultipleFileStatusInputType)
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala:25]: Should this be re-named: `MultipleFileStatusesInputArg`

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala [hunk 4]
```diff
@@ -28,6 +31,13 @@ object FileStatusFields {
       arguments = FileStatusInputArg :: Nil,
       resolve = ctx => ctx.ctx.fileStatusService.addFileStatus(ctx.arg(FileStatusInputArg)),
       tags = List(ValidateUserOwnsFiles(FileStatusInputArg))
+    ),
+    Field(
+      "addMultipleFileStatus",
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala:36]: Should the field be renamed: `addMultipleFileStatuses`

File: src/test/resources/json/addFileStatus_data_all.json [hunk 5]
```diff
@@ -1,9 +1,11 @@
 {
```
Comment (TomJKing) [inline src/test/resources/json/addFileStatus_data_all.json:1]: Rename file to plural: `...addFileStatuses...`?

File: src/test/resources/json/addFileStatus_mutation_alldata.json [hunk 6]
```diff
@@ -1,10 +1,14 @@
 {
```
Comment (TomJKing) [inline src/test/resources/json/addFileStatus_mutation_alldata.json:1]: Rename file to plural: `addFileStatuses...`?

File: src/test/resources/json/addFileStatus_mutation_invalid_fileid.json [hunk 7]
```diff
@@ -1,10 +1,14 @@
 {
```
Comment (TomJKing) [inline src/test/resources/json/addFileStatus_mutation_invalid_fileid.json:1]: Rename file to plural: `addFileStatuses...`?

File: src/test/resources/json/addFileStatus_mutation_not_owner.json [hunk 8]
```diff
@@ -1,10 +1,14 @@
 {
```
Comment (TomJKing) [inline src/test/resources/json/addFileStatus_mutation_not_owner.json:1]: Rename file to plural: `addFileStatuses...`?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some repetitive minor comments.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/FileStatusFields.scala:25]: I've renamed the variable and the argument name

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**